flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/BufferWrapper.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink.writer;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.Serializable;
+import java.util.Collection;
+
+/**
+ * A flexible wrapper interface for managing buffered request entries in an async sink. This allows
+ * sink implementations to define and optimize their own data structures for request buffering.
+ *
+ * @param <RequestEntryT> The type of request entries being buffered.
+ */
+@PublicEvolving
+public interface BufferWrapper<RequestEntryT extends Serializable> {
+
+    /**
+     * Adds a request entry to the buffer. Implementations can decide how to store the entry.
+     *
+     * @param entry The request entry to add.
+     * @param prioritize If true, the entry should be prioritized (e.g., retried before others).
+     */
+    void add(RequestEntryWrapper<RequestEntryT> entry, boolean prioritize);
```

Comment:
nit: the code type is RequestEntryWrapper but request entry is referred to in the javadoc.